### PR TITLE
add go1.5 to supported releases for gofmt

### LINKED ISSUE #5057: Jenkins not testing go vet or go fmt
https://ci.openshift.redhat.com/jenkins/job/merge_pull_requests_origin/3566/console
```
hack/verify-gofmt.sh
Unknown go version 'go', skipping gofmt.
hack/verify-govet.sh
Unknown go version 'go', skipping go vet.
```

### PR DESCRIPTION
**Pull request by stevekuznetsov**
fixes #5057 partially

cc @liggitt 

What is the purpose of the version check here,  by the way?

### CODE REVIEW

**Comment by liggitt** [comment]
I think it started with gofmt, which enforced different rules for different versions of go, to the point where our source couldn't satisfy both versions simultaneously. We picked the version we wanted to enforce (which was what jenkins ran), and skipped the check in other version so we could get green travis runs in 1.3, for example

**Comment by stevekuznetsov** [comment]
The version of `fmt` that shipped with `go1.5.1` doesn't seem to have any issues with the formatting decisions of the older version -  should we "support" both or just 1.5?

**Comment by mfojtik** [comment]
@stevekuznetsov we run travis for both go 1.4 and go 1.5.1 so I think we should support both.

**Comment by stevekuznetsov** [comment]
Executive decision:

We will *only* run `go fmt` on 1.5.1. That release pins the version at https://github.com/golang/go/commit/829cc349c5990ecba52141bd978a99ee1571bc08.
Since 1.4.2, there have been a number of changes. I don't think anyone wants to (or should need to) care about drift between these tools.

**Comment by stevekuznetsov** [comment]
@liggitt  This is ready

File: hack/verify-gofmt.sh [hunk 1]
```diff
@@ -21,8 +22,10 @@ cd "${OS_ROOT}"
 
 bad_files=$(find_files | xargs gofmt -s -l)
 if [[ -n "${bad_files}" ]]; then
-  echo "!!! gofmt needs to be run on the following files: "
+  echo "FAILURE: go fmt needs to be run on the following files: "
   echo "${bad_files}"
   echo "Try running 'gofmt -s -d [path]'"
   exit 1
+else
+  echo "SUCESS: go fmt found no errors"
```
Comment (liggitt) [inline hack/verify-gofmt.sh]: typo

File: hack/verify-gofmt.sh [hunk 2]
```diff
@@ -8,8 +8,9 @@ set -o pipefail
 
 GO_VERSION=($(go version))
 
-if [[ -z $(echo "${GO_VERSION[2]}" | grep -E 'go1.4') ]]; then
-  echo "Unknown go version '${GO_VERSION}', skipping gofmt."
+# require go version to be 1.5.1 to pin `go fmt` version to 829cc34.
+if [[ -z $(echo "${GO_VERSION[2]}" | grep -E 'go1.5.1') ]]; then
```
Comment (liggitt) [inline hack/verify-gofmt.sh]: at the moment, Jenkins is running on 1.4.x again... so we need to park this until jenkins is on 1.5.x again

Comment (stevekuznetsov) [inline hack/verify-gofmt.sh]: Wonderful

**Comment by liggitt** [comment]
go1.5 enabled for gofmt already

**Pull request closed without merging**